Source/SwiftLintBuiltInRules/Rules/Idiomatic/RedundantTypeAnnotationRule.swift

```diff
@@ -233,4 +240,42 @@ private extension ExprSyntax {
             []
         }
     }
+
+    func hasRedundantLiteralType(_ type: TypeSyntax) -> Bool {
+        kind.isLiteralExpression() && type.trimmedDescription == kind.compilerInferredLiteralType 
```

Comment:
```suggestion
        type.trimmedDescription == kind.compilerInferredLiteralType 
```

This is actually enough because comparison to `nil` fails anyway.

Reply:
Great catch!